### PR TITLE
Splash page infinite scroll + adjustments

### PR DESCRIPTION
**Pull request by kryskimmel**
- Incorporated infinite scrolling to splash page. After scrolling past 10 spot tiles, a fetch request to the api handling getting all of the spots would be made to load the next batch of 10 spot tiles. The number of spot tiles to fetch at a time can be adjusted
- Heart icon no longer flickers when pressed on and immediately switches colors between #FF385B and #535350. Functionality is still needed for the heart icon to add a spot to the user's wishlist
- Adjusted the alignment of splash page items by applying horizontal margins. Items should be fully visible when viewing the splash page on an iPhone 15
- Adjusted alignment of items in the Search component as the filter icon was clipped when splash page was viewed on an iPhone 15

### CODE REVIEW

File: mobile/package.json
Comment (AnthonyBronca) [inline mobile/package.json:1]: did we want to change the script to be "expo start -`-tunnel`"?
did we want to remove expo-splash-screen?
did we want to remove react-native-file-access?

File: mobile/screens/Home/Components/Spots.tsx
Comment (AnthonyBronca) [inline mobile/screens/Home/Components/Spots.tsx:1]: Did we want to move the fetch here out of redux? We could still incorporate the infinite scroll within the redux. We would just need the infinite scroll to call the redux dispatch function with an incrementing page state. If we fetch within the component like this, we may not have access to the redux store items we may need (although we may not need redux)

**Review by AnthonyBronca — CHANGES_REQUESTED** [review]
Added some comments on possible things to look into before merging. Let me know if you want to go ahead with the merge though!

File: mobile/package.json
Comment (kryskimmel) [reply mobile/package.json:1]: - [x] Change script to be "expo start --tunnel"
- [x] Remove expo-splash-screen
- [x] Remove react-native-file-access

File: mobile/screens/Home/Components/Spots.tsx
Comment (kryskimmel) [reply mobile/screens/Home/Components/Spots.tsx:1]: - [x] No longer fetching spots within component. Fetching is being done in /mobile/src/store/spots.ts using RTK Query.

File: mobile/screens/Home/Components/SearchModal.tsx
Comment (AnthonyBronca) [inline mobile/screens/Home/Components/SearchModal.tsx:1]: in a later push, we should probably aim to use "Image" from "expo-image" as it is faster and has caching. Although this should be fine for now!

File: mobile/store/spots.ts
Comment (AnthonyBronca) [inline mobile/store/spots.ts:1]: On line 20, we have a type for this called Spots located in the redux.d.ts file. We should refactor that type in a later commit

**Review by AnthonyBronca — APPROVED** [review]

**Review by kryskimmel — COMMENTED** [review]
Comments have been reviewed and changes will be made in next PR.

**Pull request merged**